src/WebJobs.Script/Binding/Http/HttpRouteFactory.cs

```diff
@@ -24,10 +26,15 @@ public IDirectRouteBuilder CreateRouteBuilder(string routeTemplate)
             return _routeFactoryContext.CreateBuilder(routeTemplate);
         }
 
-        public IHttpRoute AddRoute(string routeName, string routeTemplate, HttpRouteCollection routes)
+        public IHttpRoute AddRoute(string routeName, string routeTemplate, IEnumerable<HttpMethod> methods, HttpRouteCollection routes)
         {
             var routeBuilder = CreateRouteBuilder(routeTemplate);
-            var httpRoute = routes.CreateRoute(routeBuilder.Template, routeBuilder.Defaults, routeBuilder.Constraints);
+            var constraints = routeBuilder.Constraints;
+            if (methods != null && methods.Count() > 0)
+            {
+                constraints.Add("httpMethod", new HttpMethodConstraint(methods.ToArray()));
```

Comment:
This change allows the route collection to take http method into account when matching routes